src/main/java/tracko/model/order/Order.java

```diff
@@ -121,6 +125,32 @@ public void setDelivered() {
         this.isDelivered = true;
     }
 
+    public boolean containsItem(InventoryItem inventoryItem) {
+        return itemList.stream().anyMatch(pair -> pair.getItem().equals(inventoryItem));
+    }
+
+    /**
+     * Calculates the total price of a customer's ordered items.
+     */
+    public Double calculateTotalOrderPrice() {
+        double totalOrderPrice = 0;
+        for (int i = 0; i < itemList.size(); i++) {
+            ItemQuantityPair pair = itemList.get(i);
+            totalOrderPrice += pair.calculatePrice();
+        }
+        return totalOrderPrice;
+    }
+
+    /**
+     * Returns a copy of this {@code Order}, that holds immutable copies of its list of ordered {@code Item}s.
+     * @return A copy of this {@code Order}, that holds immutable copies of its list of ordered {@code Item}s.
+     */
+    public Order getImmutableItemsCopy() {
```

Comment:
this is a really small concern but would `getOrderWithImmutableItems()` be a more intuitive method name